### PR TITLE
feat(stale-and-close): add new options to change the days before close 

### LINKED ISSUE #214: Support different number of days for PRs and issues
I'm currently using Probot Stale and would like to migrate to this action. One thing which doesn't seem to be possible to configure here is different number of days for PRs and issues.

I can see two approaches that would work for me:

* Add `issue-days-before-stale`/`pr-days-before-stale` and `issue-days-before-close`/`pr-days-before-close` settings to override `days-before-stale` and `days-before-close`
* Add `process-issue`/`process-pr` boolean flags to disable processing of given types and I'd configure two action runs

I think second approach is better as that makes it easy to diverge any setting for PR and issues while not making the action more complex with every single such diversion.

### LINKED ISSUE #223: The inputs schema is not respected
Based on the inputs schema inside the `action.yml` file, there are two errors:
- The required value is missing
- The default value should only be a string (no number/boolean)

I will provide a PR to fix it.

### PR DESCRIPTION
**Pull request by C0ZEN**
Closes #214 
Closes #223

### CODE REVIEW

**Comment by mattklein123** [comment]
Any chance of getting this reviewed and merged? We are migrating from probot and miss this feature. Thank you so much.

**Comment by alyssawilk** [comment]
ping!  We've got a PR out to work around this but it would be lovely to have native support.

**Review by richardst89 — APPROVED** [review]
Love it! When can we start using it?

**Comment by C0ZEN** [comment]
@hross if you are not available to maintain it I am willing to FYI. Just need the rights to merge.

**Comment by hross** [comment]
@C0ZEN this looks good to me and I will merge it once you rebase.

File: src/IssueProcessor.ts [hunk 1]
```diff
@@ -139,10 +154,22 @@ export class IssueProcessor {
...
+        ? this._getDaysBeforePrStale()
+        : this._getDaysBeforeIssueStale();
+
+      if (isPr) {
+        core.info(`Days before pull request stale: ${DAYS_BEFORE_STALE}`);
+      } else {
+        core.info(`Days before issue stale: ${DAYS_BEFORE_STALE}`);
+      }
+
+      const SHOULD_MARK_WHEN_STALE: boolean = shouldMarkWhenStale(
```
Comment (hross) [inline src/IssueProcessor.ts:168]: nit: can we change these const declarations to not be all caps and follow the same standard as the other consts? (camel case).

File: src/enums/issue-type.enum.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// @todo fix this "false" report from ESLint
+// Or I just missed something IDK
+// eslint-disable-next-line no-shadow
+export enum IssueTypeEnum {
```
Comment (hross) [inline src/enums/issue-type.enum.ts:4]: nit: I'm not a fan of the all caps style. can we make this match the typescript handbook style? https://www.typescriptlang.org/docs/handbook/enums.html

**Comment by C0ZEN** [comment]
I will handle the rebase and naming conventions tomorrow.

File: README.md [hunk 3]
```diff
@@ -28,24 +28,28 @@ $ npm test
...
-|     `exempt-pr-labels`      |               Labels on the pr exempted from being marked as stale.               | Optional |
-|        `only-labels`        |                      Only labels checked for stale issue/pr.                      | Optional |
-|    `operations-per-run`     |           Maximum number of operations per run. \*Defaults to **30\***            | Optional |
-| `remove-stale-when-updated` | Remove stale label from issue/pr on updates or comments. \*Defaults to **true\*** | Optional |
-|        `debug-only`         |                   Dry-run on action. \*Defaults to **false\***                    | Optional |
-|         `ascending`         |                 Order to get issues/pr. \*Defaults to **false\***                 | Optional |
-| `skip-stale-issue-message`  |        Skip adding stale message on stale issue. \*Defaults to **false\***        | Optional |
-|   `skip-stale-pr-message`   |         Skip adding stale message on stale pr. \*Defaults to **false\***          | Optional |
+|     `days-before-issue-stale`  | Idle number of days before marking an issue as stale (override `days-before-stale`). | Optional
```
Comment (C0ZEN) [inline README.md:31]: Fix the broken syntax

**Comment by C0ZEN** [comment]
@hross FYI it is ready IMO.
I included a bit more content regarding the linter because I faced some issues with the table formatting and the enum but it's all for the better IMO.
I created small commits so you can review them easily.

**Review by hross — APPROVED** [review]

**Pull request merged**